.github/workflows/main.yml

```diff
@@ -9,75 +9,63 @@ on:
     branches:
       - '**'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  build_and_test:
+  build_core:
+    name: Build Stencil
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+      - uses: ./.github/workflows/actions/build-core
+
+  format:
+    name: Format Check
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+      - uses: ./.github/workflows/actions/format
```

Comment:
The GitHub log viewer doesn't play well with composite actions and makes the output a lot less comprehensible. It probably comes down to personal preference, but I'd rather have one large workflow with each step clearly visible in the log. The composite actions are useful when reused in multiple steps.

![Screen Shot 2022-04-15 at 1 26 08 PM](https://user-images.githubusercontent.com/714118/163601900-3f99db88-59c1-4498-a996-fa5eafa4e991.png)

![Screen Shot 2022-04-15 at 1 26 02 PM](https://user-images.githubusercontent.com/714118/163601898-cc6b98fc-ad4e-46a8-8502-84604170645e.png)

Reply:
That's fair, I tend to think it's all the same log at the end of the day. IMO I prefer the speed at the cost of readability here.  Maybe there's something that can be done here with >1 workflow and dependent workflows. That may give us a good balance if we can get each test (analysis, unit, e2e) in their own workflow. Let me play with it in a "throwaway PR" - I tried on JEDI days but couldn't get it working the way I wanted (at all 😆)